trieste/acquisition/sampler.py

```diff
@@ -61,33 +61,77 @@ def sample(self, at: TensorType) -> TensorType:
         """
 
 
-class DiscreteThompsonSampler(DiscreteSampler):
+class ThompsonSampler(Sampler, ABC):
     r"""
-    This sampler provides approximate Thompson samples of the objective function's
+    A :class:`ThompsonSampler` samples either the minimum values or minimisers of a function
+    modelled by an underlying :class:`ProbabilisticModel` across a  discrete set of points.
+
+    """
+
+    def __init__(self, sample_size: int, model: ProbabilisticModel, sample_min_value: bool = False):
+        """
+        :param sample_size: The desired number of samples.
+        :param model: The model to sample from.
+        :sample_min_value: If True then sample from the minimum value of the function,
+            else sample the function's minimiser.
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
+        """
+        super().__init__(sample_size, model)
+        self._sample_min_value = sample_min_value
+
+    def __repr__(self) -> str:
+        """"""
+        return f"""{self.__class__.__name__}(
+        {self._sample_size!r},
+        {self._model!r},
+        {self._sample_min_value})
+        """
+
+    @abstractmethod
```

Comment:
You don't need to repeat the abstract method here as it's already inherited from `Sampler` (Python will figure out that this must be an abstract base class as it doesn't implement it).